eos-update-flatpak-repos.service

```diff
@@ -24,6 +24,11 @@ After=eos-fix-flatpak-branches.service
 Requires=eos-extra-settled.target
 After=eos-extra-settled.target
 
+# In lieu of proper locking, we must not run at the same time as anything
+# else which touches flatpak configuration.
+Conflicts=eos-updater.service
```

Comment:
Does this work? Since this service has `RemainAfterExit=true`, then it seems like `eos-updater.service` won't be able to run if this one has started.

Reply:
From my testing, it does seem to work — systemd will `SIGTERM` the service which is already running if you explicitly start the other one. D-Bus activation does seem to count as ‘explicitly starting’ `eos-updater.service`. However, you’ve identified the bit of this patch which I’m the most wary of. Further testing (on a system which isn’t dev-converted) would definitely not go amiss.

Reply:
I’m going to fire up a non-converted eos3.3 image and test this there.

Reply:
Seems to work as expected on a non-converted eos3.3 system ([log here](https://phabricator.endlessm.com/T22525#584521)). Please do check over the log and make sure I’m not missing anything or making a thinko.

Reply:
Note that starting `eos-updater.service` by calling `eos-updater-ctl poll` is entirely equivalent to what `eos-autoupdater` does (from systemd’s point of view): `eos-autoupdater` calls D-Bus methods on the updater to start it. The alternative (which would potentially interact differently with `Conflicts=`) would be for `eos-updater.service` to have its own `eos-updater.timer` unit which triggered it. We’d then need to investigate how systemd timers interacted with `Conflicts=`.

That is not the case though: the timer is for `eos-autoupdater.service`.